action.yml

```diff
@@ -51,103 +51,58 @@ runs:
       env:
         REDMINE_DIR: ${{ inputs.path }}
 
-    - name: Install dependencies
+    - name: Check if the Redmine version is supported
       run: |
-        sudo apt-get update; \
-        sudo apt-get install -y --no-install-recommends \
-          build-essential \
-          bzr git mercurial subversion cvs \
-          ghostscript \
-          gsfonts \
-          imagemagick libmagick++-dev \
-          libnss3-dev;
+        if [ $redmine_version_number -lt $MINIUMUM_SUPPORTED_REDMINE_VERSION ]; then
+          echo "Redmine version $REDMINE_VERSION is not supported."
+          exit 1
+        fi
       shell: bash
+      env:
+        MINIUMUM_SUPPORTED_REDMINE_VERSION: "420" # v4.2.0
 
-    - name: Allow ImageMagick to read PDF files
+    - name: Set up base environment
       run: |
-        sudo sed -ri 's/(rights)="none" (pattern="PDF")/\1="read" \2/' /etc/ImageMagick-6/policy.xml
-      shell: bash
+        major_version_with_branch=$REDMINE_VERSION_MAJOR.$REDMINE_VERSION_MINOR.$REDMINE_VERSION_BRANCH
 
-    - name: Set up Google Chrome for system tests
-      run: |
-        wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
-        echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" | sudo tee /etc/apt/sources.list.d/google-chrome.list
-        sudo apt-get update
-        sudo apt-get install -y --no-install-recommends google-chrome-stable
+        case $major_version_with_branch in
+          5.1.devel)
+            setup_version="next";;
+          5.1*|5.0*|4.2*)
+            setup_version="$REDMINE_VERSION_MAJOR.$REDMINE_VERSION_MINOR";;
+          *)
+            setup_version="next";;
+        esac
+
+        echo "Running scripts/setup-base-$setup_version.sh for Redmine $REDMINE_VERSION ..."
```

Comment:
This will output like below:

```
Running scripts/setup-base-4.2.sh for Redmine 4.2.11.stable ...
```
https://github.com/hidakatsuya/action-setup-redmine/actions/runs/9920802272/job/27408068744